packages/tests/e2e/robot/tests/api/API_training_packaging.robot

```diff
@@ -98,12 +100,80 @@ Get logs of packaging
 Delete Model Trainings and Check that Model Training do not exist
     [Tags]                      training
     [Documentation]             delete model trainings
+    [Teardown]                  Cleanup resource  training  ${TRAIN_MLFLOW_DEFAULT}
     Command response list should contain id  training  ${TRAIN_MLFLOW_DEFAULT}
     Call API                    training delete  ${TRAIN_MLFLOW_DEFAULT}
     Command response list should not contain id  training  ${TRAIN_MLFLOW_DEFAULT}
 
 Delete Model Packaging and Check that Model Packaging does not exist
     [Tags]                      packaging
+    [Teardown]                  Cleanup resource  packaging  ${PACKAGING}
     Command response list should contain id  packaging  ${PACKAGING}
     Call API                    packaging delete  ${PACKAGING}
     Command response list should not contain id  packaging  ${PACKAGING}
+
+#############################
+#    NEGATIVE TEST CASES    #
+#############################
+
+#  TRAINING
+#############
+Try Create Training that already exists
+    [Tags]                      training  negative
+    [Setup]                     Cleanup resource  training  ${TRAIN_MLFLOW_DEFAULT}
+    [Teardown]                  Cleanup resource  training  ${TRAIN_MLFLOW_DEFAULT}
+    Call API                    training post  ${RES_DIR}/valid/training.mlflow.default.yaml
+    ${EntityAlreadyExists}      Format EntityAlreadyExists  ${TRAIN_MLFLOW_DEFAULT}
+    Call API and get Error      ${EntityAlreadyExists}  training post  ${RES_DIR}/valid/training.mlflow.default.update.yaml
+
+Try Update not existing and deleted Training
+    [Tags]                      training  negative
+    ${WrongHttpStatusCode}      Format WrongHttpStatusCode  ${TRAIN_NOT_EXIST}
+    Call API and get Error      ${WrongHttpStatusCode}  training put  ${RES_DIR}/invalid/training.update.not_exist.json
+    ${WrongHttpStatusCode}      Format WrongHttpStatusCode  ${TRAIN_MLFLOW_DEFAULT}
```

Comment:
It is more common to check one thing per one test. I try to stick to this rule. A lot of tests in this PR have 2 independent assertions. Maybe it is a normal practice in E2E, I don't know. But I had written this, I would have split these tests into 2 separate ones.

Reply:
OK, I will rewrite it, to stick to the best practices.

Reply:
fixed